CEGameApp.API/Data/AuthRepository.cs

```diff
@@ -0,0 +1,79 @@
+using System;
+using System.Threading.Tasks;
+using CEGameApp.API.Models;
+using Microsoft.EntityFrameworkCore;
+
+namespace CEGameApp.API.Data
+{
+    public class AuthRepository : IAuthRepository
+    {
+        private readonly DataContext _context;
+
+        public AuthRepository(DataContext context)
+        {
+            _context = context;
+        }
+
+        public async Task<Users> Login(string userName, string password)
+        {
+            var user = await _context.Users.FirstOrDefaultAsync(x => x.Name == userName);
+
+            if(user == null)
+                return null;
+            
+            if(!VerifiedPasswordHash(password, user.PasswordHash, user.PasswordSalt))
+                return null;
+
+            return user;
+        }
+
+        private bool VerifiedPasswordHash(string password, byte[] passwordHash, byte[] passwordSalt)
+        {
+            //from IDisposable, will dispose of the resources in curly braces once done with them.
+            //passing passwordSalt to generate key
+            using (var hmac = new System.Security.Cryptography.HMACSHA512(passwordSalt))
```

Comment:
hmac? maybe shaHash or evevnhMACHShaFiveTwelveHash

Reply:
Not enough info on those, and for demo purposes this should work.